### PR TITLE
Attempt to fix slow precompiled header test compilation time

### PR DESCRIPTION
**Pull request by rcurtin**
Over the past few weeks, I've been noticing that `mlpack_test` sure does take a long time to compile...

```
$ time make -j12 mlpack_test
real	23m24.389s
user	251m22.569s
sys	7m47.206s
```

On a hunch, I wondered if it was the precompiled header changes.  So I disabled them.  Then I saw:

```
$ time make -j12 mlpack_test
real	5m34.742s
user	60m44.426s
sys	1m41.422s
```

Wow!  I was surprised.  I spent more time playing around with it, and my conclusion is that at least with GCC, precompiling `mlpack.hpp` slowed down compilation of `mlpack_test` significantly, because `mlpack.hpp` is a huge header that isn't actually needed by virtually any of the tests.  So it's actually harmful to add to the precompiled header file, since it just makes it larger, and then the compiler has to do more work filtering out unnecessary code.  At least, that is my belief---I could be wrong, but the tuned numbers don't lie:

```
$ time make -j12 mlpack_test
real	4m34.066s
user	47m35.440s
sys	1m47.658s
```

Let's see what happens with the Windows build.  I think that build is the reason precompiled headers got enabled in the first place in this way...

### CODE REVIEW

File: src/mlpack/base.hpp [hunk 1]
```diff
@@ -12,13 +12,6 @@
 #ifndef MLPACK_BASE_HPP
 #define MLPACK_BASE_HPP
 
-// First, check if Armadillo was included before, warning if so.
-#ifdef ARMA_INCLUDES
-#pragma message "Armadillo was included before mlpack; this can sometimes cause\
- problems.  It should only be necessary to include <mlpack/core.hpp> and not \
-<armadillo>."
-#endif
-
```
Comment (shrit) [inline src/mlpack/base.hpp:21]: and finally

File: src/mlpack/tests/main.cpp [hunk 2]
```diff
@@ -8,10 +8,7 @@
  * 3-clause BSD license along with mlpack.  If not, see
  * http://www.opensource.org/licenses/BSD-3-Clause for more information.
  */
-#ifndef MLPACK_ENABLE_ANN_SERIALIZATION
-  #define MLPACK_ENABLE_ANN_SERIALIZATION
```
Comment (shrit) [inline src/mlpack/tests/main.cpp:12]: are we doing this manually now? would not this affect the examples?

**Review by shrit — APPROVED** [review]

File: src/mlpack/tests/main.cpp (hunk 2 above)
Comment (rcurtin) [reply src/mlpack/tests/main.cpp:12]: No, it's still necessary to define `MLPACK_ENABLE_ANN_SERIALIZATION` for serializing neural networks, but this isn't needed for `main.cpp`, which is in a separate translation unit from the other `.cpp` files.  Instead it is only needed directly in the files that test neural network serialization.  This helps speed up compilation of those files that don't need neural network serialization.

**Review by shrit — APPROVED** [review]

**Comment by rcurtin** [comment]
Looks like the Windows build runner has been updated in a broken way that requires a workaround:
https://github.com/actions/runner-images/issues/10004

I incorporated a workaround here and hope to remove it whenever Github fixes the issue.

**Pull request merged**